tango/workspace.py

```diff
@@ -419,6 +419,14 @@ def step_result(self, step_name: str) -> Any:
                 return self.step_cache[run.steps[step_name]]
         raise KeyError(f"No step named '{step_name}' found in previous runs")
 
+    @abstractmethod
+    def step_cache_remove(self, step_unique_id: str):
```

Comment:
Let's just call this `remove_step`. Since we are not only removing the step from the cache, but also from step info, etc. Workspace object user does not need to know how/where steps are stored. All they care about is removing the step from the workspace.